app/daemon/daemon.go

```diff
@@ -112,105 +117,196 @@ func DownloadAndStart(serverConfig *conf.ServerConfig, userConfig *conf.Config)
 }
 
 
-func start(daemonPath string) (*exec.Cmd)  {
-	
-	log.Println("Booting deamon")
-	cmd := exec.Command(daemonPath)
-	cmd.Start()
-
-	return cmd
-
-}
-
 func Stop(cmd *exec.Cmd) {
 
 	if err := cmd.Process.Kill(); err != nil {
 		log.Fatal("failed to kill: ", err)
 	}
 }
 
-func NewDaemonAvailable(config *conf.ServerConfig) (bool, error) {
+
+func CheckForDaemon (serverConfig *conf.ServerConfig, userConfig *conf.Config) (string, error) {
 
 	log.Println("Checking daemon")
 
 	// get the latest release info
-	releaseVersion, versionErr := getReleaseVersion(config)
-	if versionErr != nil {
-		return false, versionErr
-	}
+	releaseVersion := userConfig.RunningNavVersion
 
-	log.Println("Latest version: " + releaseVersion)
+	log.Println("Looking for version: " + releaseVersion)
 
 	// get the apps current path
 	path, err := getCurrentPath()
 	if err != nil {
-		return false, err
+		return "", err
 	}
```

Comment:
No issue here, just wanted to say that being able to return multiple things at once is sweeeeet

Reply:
So much win!